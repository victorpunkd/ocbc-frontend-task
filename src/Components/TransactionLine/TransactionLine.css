.transactionLineContainer {
  display: flex;
  padding: 5% 0%;
}

.date {
  flex: 1;
  width: 20%;
  font-weight: 700;
}

.description {
  width: 60%;
  flex: 1;
  white-space: nowrap;
  width: 100%;
  overflow: hidden;
  o-text-overflow: ellipsis;
  text-overflow: ellipsis;
}

.amount {
  width: 20%;
  flex: 1;
  text-align: right;
  font-weight: 700;
}

.greenFront {
  color: green;
}
