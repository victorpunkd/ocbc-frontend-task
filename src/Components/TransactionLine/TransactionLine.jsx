import React, { useState, useEffect, useCallback, useMemo } from "react";
import "./TransactionLine.css";

const TransactionLine = (props) => {
  const monthNames = useMemo(
    () => [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ],
    []
  );
  const [transactionDate, setTransactionDate] = useState();

  const getForamttedDate = useCallback(() => {
    let convertedDate = new Date(Date.parse(props.date));
    return `${convertedDate.getDate()} ${monthNames[convertedDate.getMonth()]}`;
  }, [monthNames, props.date]);

  useEffect(() => {
    setTransactionDate(getForamttedDate());
  }, [getForamttedDate]);

  return (
    <div className="transactionLineContainer">
      <div className="date" style={{ textAlign: "left" }}>
        {transactionDate}
      </div>
      <div className="description">{props.description || "No description"}</div>
      <div
        className={`amount ${
          props.transactionType === "receive" && "greenFront"
        }`}
      >
        {`${props.transactionType === "receive" ? "+" : "-"}${props.amount}`}
      </div>
    </div>
  );
};

export default TransactionLine;
