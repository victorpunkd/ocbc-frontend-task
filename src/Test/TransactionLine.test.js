import React from "react";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TransactionLine from "../Components/TransactionLine/TransactionLine";

configure({ adapter: new Adapter() });

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("TransactionLine", () => {
  it("snapshot renders", () => {
    const component = renderer.create(<TransactionLine />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
