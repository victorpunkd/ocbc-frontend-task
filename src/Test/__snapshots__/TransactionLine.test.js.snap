// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionLine snapshot renders 1`] = `
<div
  className="transactionLineContainer"
>
  <div
    className="date"
    style={
      Object {
        "textAlign": "left",
      }
    }
  />
  <div
    className="description"
  >
    No description
  </div>
  <div
    className="amount false"
  >
    -undefined
  </div>
</div>
`;
